### PR TITLE
feat(cli): Introduce `--data` option for `trigger sequence` command

### LINKED ISSUE #8803: keptn trigger sequence does not accept metadata
Your issue may already be reported! Please search on the [issue tracker](https://github.com/keptn/keptn/issues) before creating a new one.

**Environment**

* Keptn Version (`keptn version`): `all`

**Affected Component**

* [ ] Approval
* [ ] Bridge
* [x] CLI
* [ ] Datastore
* [ ] Docs
* [ ] Distributor / Event Broker
* [ ] Helm
* [ ] Istio
* [ ] Jmeter
* [ ] Kubernetes Integration
* [ ] Openshift Integration
* [ ] REST API
* [ ] Webhook

**Describe the bug**
The `keptn trigger sequence` command does not accept metadata so image, version or any custom metadata cannot be added.

### PR DESCRIPTION
**Pull request by warber**
closes #8803 

This PR introduces the `--data` option for the `trigger sequence` command in order to pass custom data fields contained in the `data` block of the cloude event emmitted by the CLI.

e.g.:
```bash
trigger sequence mysequence --project=ultimate-echo-project --stage=first-stage --service=myservice --data=test.strategy=direct,evaluation.timeframe=5m
```
image

Note that you cannot "overwrite" project,service,stage or labels with this.

### CODE REVIEW

File: cli/cmd/trigger_sequence.go [hunk 1]
```diff
@@ -16,13 +13,17 @@ import (
...
-	Project *string            `json:"project"`
-	Service *string            `json:"service"`
-	Stage   *string            `json:"stage"`
-	Labels  *map[string]string `json:"labels"`
+	Project  *string            `json:"project"`
+	Service  *string            `json:"service"`
+	Stage    *string            `json:"stage"`
+	Labels   *map[string]string `json:"labels"`
+	Data     *map[string]string `json:"data"`
+	DataFrom *string            `json:"data-from"`
```
Comment (odubajDT) [inline cli/cmd/trigger_sequence.go:26]: where is this used ?

Comment (warber) [reply cli/cmd/trigger_sequence.go]: tx, nowhere, deleted

**Review by odubajDT — DISMISSED** [review]

File: cli/internal/textutils_test.go [hunk 2]
```diff
@@ -27,3 +28,49 @@ func Test_JSONPathToJSONObj(t *testing.T) {
...
+			args: args{
+				toAdd: map[string]string{"project": "pr", "stage": "st", "service": "sv"},
+			},
+			want:    map[string]interface{}{"project": "pr", "stage": "st", "service": "sv"},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "valid additional fields",
+			args: args{
+				toAdd: map[string]string{"project": "pr", "stage": "st", "service": "sv", "a.b": "value", "b.a": "othervalue"},
```
Comment (thisthat) [inline cli/internal/textutils_test.go:53]: maybe also this to be extra sure: 
```suggestion
				toAdd: map[string]string{"project": "pr", "stage": "st", "service": "sv", "a.b": "value", "b.a": "othervalue", "b.a.c": "myotherCvalue"},
```

Comment (warber) [reply cli/internal/textutils_test.go:54]: as "b.a" is already bound to a value (string "othervalue") this case is not possible and marks the limit of the solution implemented here. Nonetheless I can more specifically test this "failure" case.

**Review by odubajDT — APPROVED** [review]

**Pull request merged**